channel/state/outcome/outcome.go

```diff
@@ -51,6 +52,29 @@ func (a Allocations) Total() *big.Int {
 	return total
 }
 
+// AffordsFor computes the amount that the allocations can afford for a given allocation, assuming it is funded with x coins
+func (allocations Allocations) AffordsFor(given Allocation, x *big.Int) *big.Int {
+	bigZero := big.NewInt(0)
+	affordsInTotal := big.NewInt(0).Set(bigZero)
+	surplus := big.NewInt(0).Set(x)
+	for _, allocation := range allocations {
+
+		if surplus.Cmp(bigZero) == 0 {
+			break
+		}
+
+		affords := math.BigMin(surplus, allocation.Amount)
+
+		if allocation.Equal(given) {
+			affordsInTotal = affordsInTotal.Add(affordsInTotal, affords)
```

Comment:
Should we just return `affords` here? (Or else, relax the `Equal` check to `allocation.destination == given.destination`).

As is, this method returns the total afforded to a destination if each allocation to that destination is for the same amount, but less than that if the allocations to the destination are for different amounts.

IE:
`[{Alice: 2, Bob: 3, Alice: 2}].AffordsFor({Alice:2}, 100)` returns `4`, but
`[{Alice: 2, Bob: 3, Alice: 3}].AffordsFor({Alice:2}, 100)` returns `2`

Reply:
Hmm yeah this is a case I didn't give enough thought to. It seems as though want we really want is to check against the destination, the type and the metadata but *not* the amount. In the words, accumulate the total amount flowing _in a particular way_ to _a particular destination_. 

Thinking about it from the virtual funding protocol's point of view, though -- perhaps it is better to return have a `Affords` function returning a `bool`? Then, the protocol can ask of an outcome -- do you have this exact guarantee (which it will have precomputed), and can you (completely) afford it?

Reply:
a28bc0b